networkx/classes/function.py

```diff
@@ -1288,7 +1288,9 @@ def is_path(G, path):
         True if `path` is a valid path in `G`
 
     """
-    return all((node in G and nbr in G[node]) for node, nbr in nx.utils.pairwise(path))
+    return all(
+        (node in G and nbr in G._adj[node]) for node, nbr in nx.utils.pairwise(path)
+    )
```

Comment:
Apparently, I like weeds!  
Putting the try/except into `is_path` allows us to bypass the node lookup and just run freely to the edge lookup. We have to catch KeyError and TypeError. A small speedup on a small time.  But since we're here, we might as well do it. :) 
```suggestion
    try:
        return all(nbr in G._adj[node] for node, nbr in nx.utils.pairwise(path))
    except (KeyError, TypeError):
        return False
```